package/MDAnalysis/core/topologyattrs.py

```diff
@@ -1039,11 +1039,17 @@ def principal_axes(group, pbc=False):
         """
         atomgroup = group.atoms
         e_val, e_vec = np.linalg.eig(atomgroup.moment_of_inertia(pbc=pbc))
-
+        
         # Sort
         indices = np.argsort(e_val)[::-1]
-        # Return transposed in more logical form. See Issue 33.
-        return e_vec[:, indices].T
+        # Make transposed in more logical form. See Issue 33.
+        e_vec = e_vec[:, indices].T
+
+        # Make sure the right hand convention is followed
+        if np.dot(np.cross(e_vec[0], e_vec[1]), e_vec[2]) == -1.0:
```

Comment:
Don’t check floats for exact equality. Better just check if < 0

Reply:
Sure - that makes sense, this was actually already pointed out to me by @RMeli in a private chat.